Source/Scene/ModelExperimental/UniformType.js

```diff
@@ -0,0 +1,21 @@
+var UniformType = {
+  FLOAT: "float",
+  VEC2: "vec2",
+  VEC3: "vec3",
+  VEC4: "vec4",
+  INT: "int",
+  INT_VEC2: "ivec2",
+  INT_VEC3: "ivec3",
+  INT_VEC4: "ivec4",
+  BOOL: "bool",
+  BOOL_VEC2: "bvec2",
+  BOOL_VEC3: "bvec3",
+  BOOL_VEC4: "bvec4",
+  MAT2: "mat2",
+  MAT3: "mat2",
+  MAT4: "mat4",
+  SAMPLER_2D: "sampler2D",
```

Comment:
Textures are useful to support. 

The implementation should be similar to what `Material` does with textures. (hm... I wonder if we can factor out a helper function like `Texture.fromUri() -> Promise<Texture>` or if that's not easy to do given that you need access to the Context...)

The one thing though is managing the resources. @sanjeetsuhag and I talked about this, we think the model should own the CustomShader and therefore should call `customShader.destroy()` from `ModelExperimental.destroy()`. This is similar to how the scene owns its primitives.